cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -1661,14 +1661,32 @@ ReconciliationState createReconciliationState(Reconciliation reconciliation, Kaf
         }
 
         Future<ReconciliationState> entityOperatorUserOpRoleBinding() {
+            Future ownNamespaceFuture;
+            Future watchedNamespaceFuture;
+
+            // Create role binding for the watched namespace
             String watchedNamespace = entityOperator != null && entityOperator.getUserOperator() != null ?
                     entityOperator.getUserOperator().getWatchedNamespace() : null;
-            return withVoid(roleBindingOperator.reconcile(
-                    watchedNamespace != null && !watchedNamespace.isEmpty() ?
-                            watchedNamespace : namespace,
+
+            if (watchedNamespace != null && !watchedNamespace.isEmpty() && !namespace.equals(watchedNamespace))    {
+                watchedNamespaceFuture = roleBindingOperator.reconcile(
+                        watchedNamespace,
+                        EntityUserOperator.roleBindingName(name),
+                        eoDeployment != null && entityOperator.getUserOperator() != null ?
+                                entityOperator.getUserOperator().generateRoleBinding(namespace) : null);
```

Comment:
is it `watchedNamespace`, isn't it?

Reply:
Never mind, I was wrong.

Reply:
This is the namespace of the service account -> so this should be `namespace`.